hbase-server/src/test/java/org/apache/hadoop/hbase/master/TestMaster.java

```diff
@@ -100,6 +104,25 @@ public static void afterAllTests() throws Exception {
     TEST_UTIL.shutdownMiniCluster();
   }
 
+  @Test
+  public void testRegionServerStartup() throws Exception {
+    MiniHBaseCluster cluster = TEST_UTIL.getHBaseCluster();
+    HMaster m = cluster.getMaster();
+    // Check there are 2 regionservers in master's online servers list
+    Assert.assertEquals(2, m.getServerManager().countOfRegionServers());
+    // Add invalid peer so that regionserver startup will take a long time
+    try {
+      ReplicationPeerConfig rpc = new ReplicationPeerConfig();
+      admin.addReplicationPeer("INVALID-PEER", rpc);
+    } catch (DoNotRetryIOException e) {
```

Comment:
The test also passes without the patch, did I miss something?  Not sure I fully understand the intent here, if you get an exception with add_peer RPC, its not added, right?